src/shogun/regression/LeastAngleRegression.h

```diff
@@ -13,12 +13,26 @@
 
 #include <shogun/lib/config.h>
 
+using namespace Eigen;
+using namespace std;
+
 #ifdef HAVE_LAPACK
 #include <vector>
 #include <shogun/machine/LinearMachine.h>
 
 namespace shogun
 {
+
+template <typename inputType>
+struct EigenTypes
+{
+	using ST = inputType;
+	typedef Eigen::Matrix<ST,-1,1,0,-1,1> EigenVectorXt;	
```

Comment:
i reckon this is typedef-ed in SGVector:
https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/lib/SGVector.h#L40

you could use that from there, it's a public definition.